src/block-management/block-holder.js

```diff
@@ -106,7 +108,17 @@ export default compose( [
 		} = dispatch( 'core/editor' );
 
 		return {
-			onSelect: () => {
+			onSelect: ( event ) => {
+				if ( event ) {
+					// This fix an issue with the way we are handling focus on TextIput fields and handling the keyboard state.
+					// When moving from a TextInput field to another kind of field the call that hides the keyboard was not invoked properly,
+					// resulting in keyboard up when it should not be there.
+					const currentlyFocusedTextInput = TextInputState.currentlyFocusedField();
+					if ( event.nativeEvent.target !== currentlyFocusedTextInput && ! TextInputState.isTextInput( event.nativeEvent.target ) ) {
+						// This check that the current TextInputState "focused" field is the one that has the focus.
+						TextInputState.blurTextInput( currentlyFocusedTextInput );
+					}
+				}
```

Comment:
I'm not sure this logic belongs here. The opposite (focusing on select) is handled on `RichText.componentDidUpdate`, and I think we could add the inverse there (if it was selected and now it's not, then blur). What do you think?

Reply:
I've tested your suggested approach locally @koke and it seems to work well 👍 thank you for bringing this clarity with that

Branch for Mobile Gutenberg here @daniloercoli  https://github.com/wordpress-mobile/gutenberg-mobile/compare/issue/308-call-blur-when-not-selected?expand=1

React Native Aztec:
https://github.com/wordpress-mobile/react-native-aztec/compare/add/blur-method?expand=1

And Gutenberg:
https://github.com/WordPress/gutenberg/compare/mobile...add/blur-signal-when-rich-text-deselected?expand=1

Reply:
Yes, I did exactly the same and was preparing a new PR for it.
Please open a PR then, since my code is the same of your.